packages/contracts/contracts/split/split-provider/SplitProviderV0.sol

```diff
@@ -0,0 +1,190 @@
+     * @dev Intentionally unpermissioned; no owner required; all created splits
+     * are immutable and deterministic
+     * @param splitInputs The split input parameters.
+     * @return splitter The newly created splitter contract address.
+     */
+    function getOrCreateSplitter(
+        SplitInputs calldata splitInputs
+    ) external override returns (address) {
+        // create Split struct from SplitInputs
+        ISplitFactoryV2.Split memory splitParams = _getSplitParams({
+            splitInputs: splitInputs
+        });
+        // use sender as salt to ensure unique splitters across contracts/partners for accounting
+        bytes32 salt = bytes32(bytes20(msg.sender));
+        // determine if splitter already exists
+        (address splitter, bool exists) = _splitFactoryV2.isDeployed({
+            _splitParams: splitParams,
+            _owner: address(0), // no owner, immutable,
+            _salt: salt
+        });
+        if (!exists) {
+            // create new splitter contract
+            // @dev no need to re-assign returned address; isDeployed already populated
+            _splitFactoryV2.createSplitDeterministic({
+                _splitParams: splitParams,
+                _owner: address(0), // no owner, immutable
+                _creator: address(this),
+                _salt: salt
+            });
+
+            // emit event for new splitter creation
+            emit SplitterCreated({
+                splitter: splitter,
+                splitParams: splitParams
+            });
+        }
+
+        return splitter;
+    }
+
+    /**
+     * @notice Get the immutable 0xSplits SplitFactoryV2 address used by this SplitProviderV0.
+     * @return splitFactoryV2 The 0xSplits SplitFactoryV2 address.
+     */
+    function getSplitFactoryV2() external view returns (ISplitFactoryV2) {
+        return _splitFactoryV2;
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view override returns (bool) {
+        // broadcast support of the getOrCreateSplitter function
+        // @dev intentionally only including getOrCreateSplitter, as that is the only function
+        // that is intended to be used for on-chain integration (_type is for indexing and introspection only)
+        return
+            interfaceId == this.getOrCreateSplitter.selector ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @notice Indicates the type of the contract, e.g. `SplitProviderV0`.
+     * @return type_ The type of the contract.
+     */
+    function type_() external pure returns (bytes32) {
+        return TYPE;
+    }
+
+    function _getSplitParams(
+        SplitInputs calldata splitInputs
+    ) private pure returns (ISplitFactoryV2.Split memory splitParams) {
+        // define allocations in BPS to avoid any rounding errors
+        uint256 ArtistAndAdditionalRoyaltyPercentage = splitInputs
```

Comment:
any reason why these vars are capitalized 👀 ?

Reply:
good call - fixed in [8557573](https://github.com/ArtBlocks/artblocks-contracts/pull/1455/commits/8557573a08596612c2e7277fcbc173747019b34d)